### DISCUSSION TITLE
Async send_photo, application.run_polling

### DISCUSSION THREAD

**Discussion by formulatimer** [python-telegram-bot/python-telegram-bot#4077] [Q&A] [answered]
I was using v13, and now I have rewritten the code to use v20.

1. I have noticed that the function 'send_photo' in v20 is asynchronous. Is there a similar 'send_photo' without asynchronous functionality in v20 ?

2. In v13 I used updater.start_polling() that enbabled that the code continue running, now I am using application.run_polling(allowed_updates=Update.ALL_TYPES) but the code stop at that line.  Is there a similar 'updater.start_polling()' in v20?

Thanks

**Comment by Bibo-Joshi** [comment, marked as answer]
1. v20 is asyncio-only
2. [`Application.run_polling`]() starts the polling and keeps the event loop running until you stop the process. This is roughly equivalent to the combination of v13's `updater.start_polling` + `updater.idle()`. If you want to use some custom logic for running your script, please have a look at the docs of `run_polling` as well as [this faq entry](https://github.com/python-telegram-bot/python-telegram-bot/wiki/Frequently-requested-design-patterns#running-ptb-alongside-other-asyncio-frameworks)
